dotnetv3/AutoScaling/scenarios/AutoScale_Basics/AutoScale_Basics/AutoScaleBasics.cs

```diff
@@ -0,0 +1,247 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+  Before running this SDK for .NET (v3) code example, set up your development
+  environment, including your credentials.
+
+  For more information, see the following documentation:
+
+  https://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/net-dg-setup.html
+
+  This example creates an Amazon Elastic Compute Cloud (Amazon EC2) launch
+  template. You can also use the Amazon EC2 console. For more information, see
+  the following:
+
+  https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-launch-templates.html#create-launch-template
+
+ This code example performs the following operations:
+  1. Creates an Amazon EC2 Launch Template
+  2. Creates an AWS Auto Scaling group.
+  3. Shows the details of the new AWS Auto Scaling group to show that only
+     one instance was created.
+  4. Enables metrics collection.
+  5. Updates the AWS Auto Scaling group to increase the max size to three.
+  6. Describes AWS Auto Scaling groups again to show the current state of the
+     group.
+  7. Changes the desired capacity of the AWS Auto Scaling group to two.
+  9. Retrieves the details of the group and shows the number of instances.
+ 10. Lists the scaling activities that have occurred for the group.
+ 11. Displays the Amazon CloudWatch metrics that have been collected.
+ 12. Terminates an instance in the Auto Scaling group.
+ 13. Disables metrics collection.
+ 13. Deletes the Auto Scaling group.
+ 14. Deletes the EC2 launch template.
+*/
+
+// snippet-start:[AutoScale.dotnetv3.AutoScale_Basics.main]
+var imageId = "ami-05803413c51f242b7";
+var instanceType = "t2.micro";
+var launchTemplateName = "AutoScaleLaunchTemplate";
+
+// The name of the Auto Scaling group.
+var groupName = "AutoScaleExampleGroup";
+
+// the Amazon Resource Name (ARN) of the service linked IAM role.
+var serviceLinkedRoleARN = "<Enter Value>";
+
+var client = new AmazonAutoScalingClient(RegionEndpoint.USEast2);
+
+Console.WriteLine("Auto Scaling Basics");
+DisplayDescription();
+
+// Start by creating the and save launch template Id to use when deleting the launch template at
+// the end of the application.
```

Comment:
// Create the launch template and save the template Id to use when deleting the launch template at
// the end of the application.

Reply:
fixed